packages/kbn-esql-ast/src/types.ts

```diff
@@ -25,18 +27,24 @@ export interface ESQLLocation {
   max: number;
 }
 
-interface ESQLAstBaseItem {
-  name: string;
+export interface ESQLAstBaseItem<Name = string> {
+  name: Name;
```

Comment:
Maybe in the future we can rename this field to `kind`, i.e. *the type of a type*.

Reply:
Agreed. 👍

Reply:
Added to https://github.com/elastic/kibana/discussions/182393#discussioncomment-9296962